### PR TITLE
[Build/TF-Lite] Check tf-lite version and don't use contrib path @open sesame 9/24 16:03

### PR DESCRIPTION
**Pull request by dongju-chae**
This patch adds tf-lite version checking and makes it remove
'contrib' include path in tflite subplugin.

It resolves https://github.com/nnstreamer/nnstreamer/issues/2756.

Signed-off-by: Dongju Chae <dongju.chae@samsung.com>

**Self evaluation:**
1. Build test: [ ]Passed [ ]Failed [*]Skipped
2. Run test: [ ]Passed [ ]Failed [*]Skipped

### CODE REVIEW

File: ext/nnstreamer/tensor_filter/meson.build [hunk 1]
```diff
@@ -101,20 +106,38 @@ if tflite_support_is_available
...
-  endif
-  if cxx.has_type('kTfLiteFloat16', prefix : '#include <tensorflow/contrib/lite/model.h>')
-    tflite_compile_args += '-DTFLITE_FLOAT16=1'
-  endif
-  if cxx.has_type('kTfLiteComplex64', prefix : '#include <tensorflow/contrib/lite/model.h>')
-    tflite_compile_args += '-DTFLITE_COMPLEX64=1'
+  # Since tf-1.13, tflite has moved from contrib to core
+  if tflite_versions[1].to_int() >= 13
+    tflite_compile_args += '-DTFLITE_NO_CONTRIB'
+    if cxx.has_type('kTfLiteInt8', prefix : '#include <tensorflow/lite/model.h>')
```
Comment (jaeyun-jung) [inline ext/nnstreamer/tensor_filter/meson.build:115]: is it possible to set header prefix and remove duplicated lines?
```
tflite_h_prefix='#include <tensorflow/lite/model.h>'
if tflite_versions[1].to_int() < 13
tflite_h_prefix='#include <tensorflow/contrib/lite/model.h>'
endif

if cxx.has_type('kTfLiteInt8', prefix : tflite_h_prefix)
  tflite_compile_args += '-DTFLITE_INT8=1'
endif
....
```

Comment (dongju-chae) [reply ext/nnstreamer/tensor_filter/meson.build:115]: Yes, looks better

File: ext/nnstreamer/tensor_filter/meson.build [hunk 2]
```diff
@@ -101,20 +106,27 @@ if tflite_support_is_available
   nnstreamer_filter_tflite_deps = tflite_support_deps + [thread_dep, libdl_dep, glib_dep, gst_dep, nnstreamer_dep]
 
   tflite_compile_args = []
+  tflite_compile_args += '-DTFLITE_VERSION=1'
```
Comment (jaeyun-jung) [inline ext/nnstreamer/tensor_filter/meson.build:109]: Discussion
TFLITE_VERSION='1.13.1'
TFLITE_MAJOR_VERSION=1
TFLITE_MINOR_VERSION=13
TFLITE_MICRO_VERSION=1

how about changing all version definition?

Comment (dongju-chae) [reply ext/nnstreamer/tensor_filter/meson.build:109]: OK, I'll update the PR.

Comment (wooksong) [reply ext/nnstreamer/tensor_filter/meson.build]: There is ```'-DTFLITE_VER``` in ext/nnstreamer/tensor_filter/meson.build.
```python
    tflite_ver_dep = declare_dependency (
      compile_args : ['-DTFLITE_VER="' + tflite_support_deps[0].version() +'"', ],
    )
```

How about generalizing this compile_args?

Comment (dongju-chae) [reply ext/nnstreamer/tensor_filter/meson.build]: Oh, I didn't see that. I will reuse or revise this.

**Review by anyj0527 — APPROVED** [review]
LGTM 👍

**Review by myungjoo — APPROVED** [review]

**Pull request merged**